### ISSUE TITLE
Add support for requesting a CAPTCHA during registration

### ISSUE THREAD

**Issue by golf1052** [signal-csharp/Signal-Windows#218]
If a user gets blocked by a 402 error during registration this means that they need to complete a CAPTCHA to resolve the error. Signal uses https://signalcaptchas.org/registration/generate.html

See https://github.com/signalapp/Signal-Android/commit/02b0800b22e2faaa1f659d34ac3bb2f44eda3631

**Comment by golf1052** [comment]
Sending CAPTCHA tokens to Signal works, the current problem I'm blocked by is that the URI generated by the Signal CAPTCHA page on https://signalcaptchas.org/registration/generate.html isn't a valid URI. This means when embedding the CAPTCHA webpage into the app I cannot handle any Navigation event correctly because the Windows APIs expect a valid URI. This also goes for URI protocol handling.

I've made a post on the [Signal community forum](https://community.signalusers.org/t/request-to-make-signalcaptchas-org-redirect-a-valid-uri/28066). Hopefully Signal will consider making a change to their page. In the mean time I'll continue looking for a work around.

**Comment by golf1052** [comment]
I've figured out an annoying but working workaround. Though the C# Uri constructor fails to parse the `signalcaptcha` Uri, both a C++ or a Javascript UWP app can parse it fine (or at least don't throw an exception). With this knowledge here is the workaround.

1. Create a Javascript UWP app and register `signalcaptcha` as a protocol. The Javascript app is [here](https://github.com/signal-csharp/SignalCaptcha).
2. When the Javascript app gets launched by the protocol, extract the token from the Uri and pass the token back to Signal-Windows. There are [many ways](https://docs.microsoft.com/en-us/windows/uwp/communication/interprocess-communication) to share data between UWP apps but I've opted to use [LaunchUriAsync](https://docs.microsoft.com/en-us/uwp/api/windows.system.launcher.launchuriasync?view=winrt-15063).
3. Since Signal-Windows is already open, when handling the protocol launch update the CAPTCHA token on the RegistrationPage.

The one major annoyance is that the user needs to install another app to get CAPTCHAs to work but I haven't found an in-app way of getting the `signalcaptcha` redirect to work yet.
